apps/fipsinstall.c

```diff
@@ -0,0 +1,350 @@
+/*
+ * Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <string.h>
+#include "apps.h"
+#include "progs.h"
+#include <openssl/evp.h>
+#include <openssl/provider.h>
+
+#define BUFSIZE 4096
+#define DEFAULT_MAC_NAME "HMAC"
+
+/* Configuration file values */
+#define FIPS_SECTION "[fips_check_section]"
```

Comment:
This will be problematic as soon as there's more than one FIPS-certified module...  Sure, so far we only plan on our own, but do we want to make a general solution for all, or will everyone have to roll their own FIPS installation solution?

Reply:
Good idea. Will make the section an optional input.